### ISSUE TITLE
Testing with mocked requests

### ISSUE THREAD

**Issue by shaps80** [maticzav/swift-graphql#183]
@maticzav this is really a question for you to begin with.

I had trouble (as have others) running the tests since we don't seem to be able to get the local server running. 

However I wonder why we even need to take that approach? I was thinking making use of URLProtocol would be far simpler and make it easier to run tests, directly from Xcode.

Can you comment on this, as perhaps I'm missing the point here ;)

**Comment by maticzav** [comment]
Yes!

I am totally for having a mock interface and doing it right.

The way I've done it in the past is hosting a server and then using that but Render became too expensive at one point and I had to shut it down.

**Comment by shaps80** [comment]
Yeah, in general, it's a far more involved approach to use a server. I'll take a look into this then. Thanks. Appreciate the feedback a lot
